llvm/lib/Target/Sparc/SparcISelLowering.cpp

```diff
@@ -1815,7 +1817,8 @@ SparcTargetLowering::SparcTargetLowering(const TargetMachine &TM,
   setOperationAction(ISD::FREM , MVT::f32, Expand);
   setOperationAction(ISD::FMA  , MVT::f32, Expand);
   setOperationAction(ISD::CTTZ , MVT::i32, Expand);
-  setOperationAction(ISD::CTLZ , MVT::i32, Expand);
+  setOperationAction(ISD::CTLZ, MVT::i32,
+                     Subtarget->isVIS3() ? Promote : LibCall);
```

Comment:
The libcall corresponds to `_ZERO_UNDEF` version (it doesn't support zero input), hence no changes in tests.
`CTLZ` should be left set to `Expand`, it should be expanded into a zero check + libcall + select.

---
It may be worth double checking that `libgcc` provides `clzsi2`. It may so happen that it only provides 64-bit version (`clzdi2`) on 64-bit Sparc.
`compiler-rt` should provide both, but if I'm not mistaken it is not the default compiler support library?

Reply:
But it doesn't produce `clzsi2` on a 64-bit system, and this may be an issue if there is no such function.

Reply:
I mean, we can use libcall for i64 on both 32- and 64-bit system,
but it appears we can't use i32 libcall on 64-bit system. At least this looks so from the godbolt link.